File: benchmark/split.py
import sys, re, os

base = 2000

str = "Newcastle Knights are the best there is"

print("space split", file=sys.stderr)

for i in range(base):
  S = str.split(" ")
  x = len(S)
  for j in range(x):
    s = S[j]

print("single char", file=sys.stderr)

for i in range(base):
  S = str.split("a")
  x = len(S)
  for j in range(x):
    s = S[j]

print("regex", file=sys.stderr)

f = open("x.tmp", "w")
for i in range(base):
  S = re.split(" |t", str)
  x = len(S)
  for j in range(x):
    s = S[j]
  f.write(str + "\n")
f.close()

print("field split", file=sys.stderr)

f = open("x.tmp", "r")
lines = f.readlines()
for S in lines:
  fields = S.split(" ")
  NF = len(fields)
  for i in range(NF):
    s = fields[i]
f.close()

os.system("rm -f x.tmp")

File: benchmark/split.pl
$base = 2000;

$str = "Newcastle Knights are the best there is";

print STDERR "space split\n";

for ($i=0; $i<$base; $i++) {
  @S = split(" ", $str);
  $x = @S;
  for ($j=1; $j<=$x; $j++) {
    $s = $S[$j];
  }
}

print STDERR "single char\n";

for ($i=0; $i<$base; $i++)
{
  @S = split("a", $str);
  $x = @S;
  for ($j=1; $j<=$x; $j++) {
    $s = $S[$j];
  }
}

print STDERR "regex\n";
open(OUTPUT, '>x.tmp');

for ($i=0; $i<$base; $i++)
{
  @S = split(/ |t/, $str);
  $x = @S;
  for ($j=1; $j<=$x; $j++) {
    $s = $S[$j];
  }
  print(OUTPUT $str,"\n");
}
close(OUTPUT);

open(INPUT, '<x.tmp');
print STDERR "field split\n";

while (<INPUT>)
{
  chop;
  @fields = split(" ");
  $NF = @fields;
  for ($i=1; $i<=$NF; $i++) {
    $s = $fields[$i];
  }
}

close(INPUT);

system("rm -f x.tmp");
